spec/integ/kicking.spec.js

```diff
@@ -88,6 +88,34 @@ describe("Kicking", function() {
             ircUserCli.emit("kick", config._chan, ircUser.nick, "KickerNick", "Reasons");
             yield kickPromise;
         }));
+        it("should not leave until all messages are sent", test.coroutine(function*() {
+            let SENT_MESSAGES = -1; // We send a message to in the beforeEach
+            const kickPromise = new Promise(function(resolve, reject) {
+                const ircUserSdk = env.clientMock._client(ircUser.id);
+                ircUserSdk.leave.and.callFake(function(roomId) {
+                    expect(roomId).toEqual(config._roomid);
+                    expect(SENT_MESSAGES).toEqual(3);
```

Comment:
I'm a little concerned about racing here, but I think it should be okay. Paranoia would be adding a lock on the variable, potentially even an order of operations check. I'm really not sure that's needed here though.